dbms/src/Functions/arrayEnumerateRanked.cpp

```diff
@@ -0,0 +1,54 @@
+#include "arrayEnumerateRanked.h"
+
+namespace DB
+{
+std::tuple<DepthType, DepthTypes, DepthType> getDepths(const ColumnsWithTypeAndName & arguments)
+{
+    const size_t num_arguments = arguments.size();
+    DepthType clear_depth = 1;
+    DepthType max_array_depth = 1;
+    DepthTypes depths;
+
+    for (size_t i = 0; i < num_arguments; ++i)
+    {
+        if (!arguments[i].column)
+            continue;
+        const auto non_const = arguments[i].column->convertToFullColumnIfConst();
```

Comment:
Const column will be converted to full here, but we only need to check it isn't an array. It's better to make `typeid_cast` or something.